QuantConnect.BybitBrokerage/BybitBaseBrokerage.Messaging.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.WebSockets;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+using QuantConnect.Brokerages;
+using QuantConnect.BybitBrokerage.Converters;
+using QuantConnect.BybitBrokerage.Models;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Logging;
+using QuantConnect.Orders;
+using QuantConnect.Orders.Fees;
+using QuantConnect.Securities;
+using QuantConnect.Util;
+using OrderStatus = QuantConnect.BybitBrokerage.Models.Enums.OrderStatus;
+
+namespace QuantConnect.BybitBrokerage;
+
+public partial class BybitBrokerage
+{
+
+    private static readonly JsonSerializerSettings Settings = new JsonSerializerSettings
+    {
+        Converters = new List<JsonConverter>() { new ByBitKlineJsonConverter(), new BybitDecimalStringConverter() },
+        NullValueHandling = NullValueHandling.Ignore,
+        ContractResolver = new DefaultContractResolver
+        {
+            NamingStrategy = new CamelCaseNamingStrategy()
+        },
+    };
+
+    // Binance allows 5 messages per second, but we still get rate limited if we send a lot of messages at that rate
+    // By sending 3 messages per second, evenly spaced out, we can keep sending messages without being limited
+    //todo
+    private readonly RateGate _webSocketRateLimiter = new RateGate(1, TimeSpan.FromMilliseconds(330));
+    protected readonly object TickLocker = new object();
+
+    
+    
+    /// <summary>
+    /// Locking object for the Ticks list in the data queue handler
+    /// </summary>
+
+    private void OnUserMessage(WebSocketMessage webSocketMessage)
+    {
+        var e = (WebSocketClientWrapper.TextMessage)webSocketMessage.Data;
+        try
+        {
+            if (Log.DebuggingEnabled)
+            {
+                Log.Debug($"{nameof(BybitBrokerage)}.{nameof(OnUserMessage)}(): {e.Message}");
+            }
+
+            var jObj = JToken.Parse(e.Message);
+            var topic = jObj.Value<string>("topic");
+            if (topic == "order")
+            {
+                var orders = JsonConvert.DeserializeObject<BybitDataMessage<BybitOrder[]>>(jObj.ToString(), Settings).Data;
+                foreach (var order in orders)
+                {
+
+                    var leanOrders = OrderProvider.GetOrdersByBrokerageId(order.OrderId).ToArray().Take(1);
+                    foreach (var leanOrder in leanOrders)
+                    {
+                        //todo why multiplied orders here? var leanOrder = leanOrders.FirstOrDefault;
+                        // test orderprovider is cloning orders, therefore the market order test never is filled and waits for an event which already fired
```

Comment:
I'm having the problem here that the orderprovider is cloning the orders while [this test](https://github.com/QuantConnect/Lean.Brokerages.ByBit/pull/1/files#diff-4dc04ec4c860de8a09761a7ae875ba6df63cc999ae8b471b6eb3028b4b50bb0aR133 ) with the market order is checking the status of the initial order object to determine if it enters the modify 
 order loop and therefore never completes as the order filled event is already submitted. Any advice there?